### PR TITLE
Test that addSingleFactorConstant knowns all unitConstants.

### PR DESCRIPTION
**Pull request by hugovdm**
Example unit test output for a fake new constant "foobar":

         testUnitConstantFreshness {
         testUnitConstantFreshness data: expected success but got error: U_INVALID_FORMAT_ERROR - addSingleFactorConstant(<foobar>, ...).
         
         If U_INVALID_FORMAT_ERROR, please check that "icu4c/source/i18n/unitconverter.cpp" has all constants? Is "foobar" a new constant?
          - (Are you missing data?)
      
         } ERRORS (1) in testUnitConstantFreshness

##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/ICU-_____
- [ ] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/test/intltest/unitstest.cpp [hunk 1]
```diff
@@ -65,6 +69,36 @@ void UnitsTest::runIndexedTest(int32_t index, UBool exec, const char *&name, cha
...
+                "If U_INVALID_FORMAT_ERROR, please check that \"icu4c/source/i18n/unitconverter.cpp\" "
+                "has all constants? Is \"%s\" a new constant?\n",
+                constant, constant)) {
+            continue;
+        }
+        // TODO(units,hugovdm): implement some symbolic maths to evaluate the
+        // values of these constants? Counter-argument: constant values don't
+        // change, and the data-driven unit tests generally take care of
+        // validating the precision of conversions, if they have enough
+        // coverage.
```
Comment (hugovdm) [inline icu4c/source/test/intltest/unitstest.cpp:98]: Question to the reviewer: Shane, I think I need not worry about the constant values, and can simply delete this TODO. Do you concur?

Comment (sffc) [reply icu4c/source/test/intltest/unitstest.cpp:98]: How about if you do something in the middle here.  If the constant can be parsed as a double, then assert that it is equal.  Otherwise, assume it's composed.  We have:

```
    unitConstants{
        G{"6.67408E-11"}
        PI{"411557987 / 131002976"}
        ft2_to_m2{"ft_to_m*ft_to_m"}
        ft3_to_m3{"ft_to_m*ft_to_m*ft_to_m"}
        ft_to_m{"0.3048"}
        gal_imp_to_m3{"0.00454609"}
        gal_to_m3{"231*in3_to_m3"}
        gravity{"9.80665"}
        in3_to_m3{"ft3_to_m3/12*12*12"}
        lb_to_kg{"0.45359237"}
    }
```

It's trivial to check:

- G
- ft_to_m
- gal_imp_to_m3
- gravity
- lb_to_kg

PI is slightly harder since it's a rational, but we can probably punt on that one since we are operating in double space.

File: icu4c/source/i18n/unitconverter.cpp [hunk 2]
```diff
@@ -318,9 +318,12 @@ void loadConversionRate(ConversionRate &conversionRate, const MeasureUnit &sourc
```
Comment (sffc) [inline icu4c/source/i18n/unitconverter.cpp:318]: Also, please update this comment:

https://github.com/icu-units/icu/blob/8948cc5ab195def83d258d7b39537b7a9e960f9e/icu4c/source/i18n/unitconverter.cpp#L89

Comment (hugovdm) [reply icu4c/source/i18n/unitconverter.cpp:322]: To me this code looks like it will populate the array every time the function is run - but I guess it's possible optimising compilers would take care of it...

I'm now proposing a C99 brace array (giving me a compiler warning), declaring the array as "static", to be sure that it gets populated only once. I am wanting to avoid six values pushed to the stack every time the function is called?

File: icu4c/source/test/intltest/unitstest.cpp (hunk 1 above)
Comment (hugovdm) [reply icu4c/source/test/intltest/unitstest.cpp]: Done.

**Comment by hugovdm** [comment]
TL;DR: Avoid GitHub's online merge functionality? GitHub is now a bit confused with this PR's state, I'm hoping it is resolved with time.

I know a heavy hand can force a Web UI update, but I've not yet managed to see evidence of a Travis run for my latest commits.

(GitHub's web-based merge / conflict-resolution functionality seemed to be stuck, I waited quite some time with no evidence of it still doing something. I then pushed a merge in the normal way from my laptop.)

**Comment by hugovdm** [comment]
Pardon the force-pushes, they were no-op, just updating the commit ID in an attempt to get Travis running. Looks like GitHub just needed time.

File: icu4c/source/i18n/unitconverter.cpp (hunk 2 above)
Comment (sffc) [reply icu4c/source/i18n/unitconverter.cpp:326]: Looks good to me, and I don't see this compiler warning on the Travis build output.
